### ISSUE TITLE
What is the last parameter for different groups other than 0 1 and 2

### ISSUE THREAD

**Issue by anikng** [petitmingchang/TO-GCN#8]
Thanks for the nice tool for comparative analysis. Can you please suggest what should be the parameter (instead of 0,1 or 2) for determining the time order in for ex, C1+C2-?

Because 0 1 and 2 can be used to analyze only 3 groups, and I am interested in other groups as well (C1+C2-, C1-C2+, C10C2+ etc)

TO-GCN 13 13 example_data/TFs_1718.tsv example_data/TFs_1718.tsv 0.84 0.84 example_data/seeds.txt **0**

**Comment by petitmingchang** [comment]
Thank you for your suggestion, I will add more options for the last parameter soon in the future.

**Comment by zjDzlll** [comment]
Hey, 
if I want to analyze the genes of other groups（ex, C1+C2-）, how do I need to adjust the parameters?

**Comment by zjDzlll** [comment]
if I analysis the genes of group(C1+C20), how do I need to adjust PCC value?

TO-GCN 13 13 example_data/TFs_1718.tsv example_data/TFs_1718.tsv 0.84 ??? example_data/seeds.txt 1

Thanks !!!

**Comment by petitmingchang** [comment]
Hi @zjDzlll and @anikng 

I just updated the original TO-GCN to TO-GCN-2 that can generate TF_levels for other condition pairs (i.e. C1+C2- and C1-C2+).
If you are interested, please download the TO-GCN-2.cpp and compile it with the command:

`g++ TO-GCN-2.cpp -o TO-GCN-2`

Different from the original TO-GCN, we need two more parameters for the negative correlation cutoff values for C1 and C2 (obtained from Cutoff and similar to the parameter setting in GCN).
Besides, we set the last parameter as 3 or 4 to get the results for condition pairs of C1+C2- or C1-C2+, respectively.
Here is an example command for analyzing C1+C2- group:

`TO-GCN-2 13 13 example_data/TFs_1718.tsv example_data/TFs_1718.tsv 0.84 0.84 -0.75 -0.75 example_data/seeds.txt 3`

petitmingchang

**Comment by zjDzlll** [comment]
Hi @petitmingchang, 
Thank you so much for your reply!
I found that select genes as seeds was very important.
So, I have some questions: I select some TFs as seeds. Does the expression patterns of these TFs need to be similar?

**Comment by petitmingchang** [comment]
Hi @zjDzlll 

Yes, it would be better to have simple one starting point in the TO-GCN analysis.
So we often use just one TF or TFs that share highly similar expression patterns in the seed list. 

petitmingchang

**Comment by zjDzlll** [comment]
Hi @petitmingchang 

After reading the manual, I still don't understand what expression patterns of seeds should choose in different groups. For example, Zm+Os0. How to select seeds?

zjDzIII

**Comment by petitmingchang** [comment]
Hi @zjDzlll 

Since, seed in a TO-GCN will be the initial node(s) for following level assignment of all other nodes. 
So, you may check your gene x time-point matrix by z-score transformation (easier for you to observe the pattern) first.
Then select one or more TF genes highly expressed at the first time point but lowly expressed at other time points in Zm data only.

petitmingchang

**Comment by zjDzlll** [comment]
Hi @petitmingchang 

If I choose a gene that is not expressed at the first time point, but the expression level of that gene gradually increases over time, can this pattern be used as a seed?

zjDzIII

**Comment by petitmingchang** [comment]
@zjDzlll 

Yes, you can use it as a seed and bit time order of GCN will be reversed.
We had adopted the similar way in this [paper](https://onlinelibrary.wiley.com/doi/full/10.1002/jmv.28478).

petitmingchang

**Comment by Wenhui-bio** [comment]
hi, I want to determined the time order in result file C1-C2-. What should I do?

**Comment by petitmingchang** [comment]
@whui2bioinformatics 

The assumption of TO-GCN is based on cascade of up-regulation among transcription factors, so we only consider positive coexpression in the time-series data (e.g. C1+C2+, C1+C20, C10C2+, etc.). It is difficult to tell the time order with both negative coexression (i.e. C1-C2-) among transcription factors. 

petitmingchang

**Comment by Wenhui-bio** [comment]
@petitmingchang 
Ok, thank you very much
